packages/stable/src/api/dataPoints/dataPointsApi.ts

```diff
@@ -114,7 +114,11 @@ export class DataPointsAPI extends BaseResourceAPI<
 
       for (const key in mergedTimeseries) {
         const item = mergedTimeseries[key];
-        item.datapoints = mergedDatapoints[key];
+        if (key in mergedDatapoints) {
+          item.datapoints = mergedDatapoints[key];
+        } else {
+          item.datapoints = [];
+        }
```

Comment:
@muradsater as a quick fix this is ok.
I do believe that we should refactor this function a bit in the future for easier readability and performance.